### PR TITLE
 BAEL-4362 - Generate WSDL Stubs with Maven

### PR DESCRIPTION
**Pull request by saleco**

### CODE REVIEW

File: maven-modules/maven-plugins/jaxws/src/main/java/com/baeldung/soap/ws/client/CountryServiceClient.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public class CountryServiceClient {
+
+    private CountryService countryService;
+
+    public CountryServiceClient(CountryService countryService) {
+        this.countryService = countryService;
+    }
+    public String getCapitalByCountryName(String countryName) {
+        return Optional.of(countryService.findByName(countryName))
+            .map(Country::getCapital).orElseThrow(CountryNotFoundException::new);
```
Comment (eric-martin) [inline maven-modules/maven-plugins/jaxws/src/main/java/com/baeldung/soap/ws/client/CountryServiceClient.java:14]: two space indent

Comment (saleco) [reply maven-modules/maven-plugins/jaxws/src/main/java/com/baeldung/soap/ws/client/CountryServiceClient.java]: Fixed

File: maven-modules/maven-plugins/jaxws/src/main/java/com/baeldung/soap/ws/client/CountryServiceClient.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.Optional;
+
+public class CountryServiceClient {
+
+    private CountryService countryService;
+
+    public CountryServiceClient(CountryService countryService) {
+        this.countryService = countryService;
+    }
+    public String getCapitalByCountryName(String countryName) {
```
Comment (KevinGilmore) [inline maven-modules/maven-plugins/jaxws/src/main/java/com/baeldung/soap/ws/client/CountryServiceClient.java:12]: add a blank line between methods

File: maven-modules/maven-plugins/jaxws/src/main/java/com/baeldung/soap/ws/client/CountryServiceClient.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        return Optional.of(countryService.findByName(countryName))
+          .map(Country::getPopulation).orElseThrow(CountryNotFoundException::new);
+    }
+
+    public Currency getCurrencyByCountryName(String countryName) {
+        return Optional.of(countryService.findByName(countryName))
+          .map(Country::getCurrency).orElseThrow(CountryNotFoundException::new);
+    }
+
+
```
Comment (KevinGilmore) [inline maven-modules/maven-plugins/jaxws/src/main/java/com/baeldung/soap/ws/client/CountryServiceClient.java:27]: remove blank lines before closing brace

File: maven-modules/maven-plugins/jaxws/src/test/java/com/baeldung/soap/ws/client/CountryServiceClientUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    @BeforeEach
+    void setUp() {
+        countryService = mock(CountryService.class);
+        countryServiceClient = new CountryServiceClient(countryService);
+    }
+
+    @DisplayName("Get capital by country name when country not found")
+    @Test
+    void getCapitalByCountryName_WhenCountryDoesNotExists_ThenShouldThrowCountryNotFoundException() {
```
Comment (KevinGilmore) [inline maven-modules/maven-plugins/jaxws/src/test/java/com/baeldung/soap/ws/client/CountryServiceClientUnitTest.java:21]: Another interpretation of the BDD-style test naming convention, and perhaps a more popular one, is to use the givenX part for the precondition, and the whenY part for the method/action being invoked.

Using this approach, this test might be named: "givenCountryDoesNotExist_whenGetCapitalByCountryName_thenThrowsCountryNotFoundException"

**Review by KevinGilmore — COMMENTED** [review]
See notes about blank lines and test naming.

**Comment by saleco** [comment]
> See notes about blank lines and test naming.

PR updated

**Pull request merged**